io/include/pcl/compression/impl/organized_pointcloud_compression.hpp

```diff
@@ -332,7 +332,14 @@ namespace pcl
 
         // reading compressed rgb data
         compressedDataIn_arg.read (reinterpret_cast<char*> (&compressedColorSize), sizeof (compressedColorSize));
-        compressedColor.resize (compressedColorSize);
+        if (compressedColorSize > 0)
```

Comment:
There's no guarantee that the variable is not more than zero on error since it is not zero initialized

Reply:
It is initialized zero elsewhere in the file

Reply:
Not really. You're reading it just in the previous line. It is declared in L295 and if the read fails, the [value is undefined (since `sizeof(compressedColorSize) > 1`)](https://en.cppreference.com/w/cpp/io/basic_istream/read)